# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../wvstreams/openssl98.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Nur wvstreams-4.0.2.orig/crypto/wvtripledes.cc wvstreams-4.0.2/crypto/wvtripledes.cc
--- wvstreams-4.0.2.orig/crypto/wvtripledes.cc	2005-03-24 04:01:23.000000000 +0200
+++ wvstreams-4.0.2/crypto/wvtripledes.cc	2005-10-23 17:31:53.000000000 +0300
@@ -102,7 +102,8 @@
 	// ECB works 64bits at a time
 	while (len >= 8)
 	{
-#if OPENSSL_VERSION_NUMBER >= 0x0090705FL
+#if OPENSSL_VERSION_NUMBER >= 0x0090705FL \
+    && OPENSSL_VERSION_NUMBER < 0x0090800FL
 	    DES_ecb3_encrypt(data, crypt,
 			     &deskey1, &deskey2, &deskey3,
 			     mode == ECBEncrypt ? DES_ENCRYPT : DES_DECRYPT);
diff -Nur wvstreams-4.0.2.orig/crypto/wvx509.cc wvstreams-4.0.2/crypto/wvx509.cc
--- wvstreams-4.0.2.orig/crypto/wvx509.cc	2005-03-24 04:01:23.000000000 +0200
+++ wvstreams-4.0.2/crypto/wvx509.cc	2005-10-23 17:41:19.000000000 +0300
@@ -1162,12 +1162,19 @@
 		else
 		{
 		    void *ext_data = NULL;
+#if OPENSSL_VERSION_NUMBER >= 0x0090800fL
+		    const unsigned char **ext_value_data;
+		    ext_value_data = (const_cast<const unsigned char **>
+				      (&ext->value->data));
+#else
+		    unsigned char **ext_value_data = &ext->value->data;
+#endif
 		    if (method->it) 
-			ext_data = ASN1_item_d2i(NULL, &ext->value->data, 
+			ext_data = ASN1_item_d2i(NULL, ext_value_data, 
 						ext->value->length, 
 						ASN1_ITEM_ptr(method->it));
 		    else
-			ext_data = method->d2i(NULL, &ext->value->data, 
+			ext_data = method->d2i(NULL, ext_value_data, 
 					      ext->value->length);
 		    
 		    if (method->i2s)
